imagersite/imager_profile/views.py

```diff
@@ -30,3 +30,26 @@ def get_context_data(self, **kwargs):
         context['private_albums'] = albums.filter(published='PRIVATE')
 
         return context
+
+
+class ProfileEditView(LoginRequiredMixin, UpdateView):
+    template_name = 'imager_profile/profile_edit.html'
+    model = ImagerProfile
+    form_class = ProfileEditForm
+    login_url = reverse_lazy('auth_login')
+    success_url = reverse_lazy('profile')
+    slug_url_kwarg = 'username'
+    slug_field = 'user__username'
+
+    def get(self, *args, **kwargs):
+        self.kwargs['username'] = self.request.user.get_username()
+        return super().get(*args, **kwargs)
+
+    def post(self, *args, **kwargs):
+        self.kwargs['username'] = self.request.user.get_username()
+        return super().post(*args, **kwargs)
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs.update({'username': self.request.user.get_username()})
+        return kwargs
```

Comment:
Since you are overwriting the `get` and `post` methods to retrieve the username, you do not need to override this method. The `UpdateView` automatically retrieves the item specified by the `slug_url_kwarg` and `slug_field`.